tests/dataframe_tests/util.py

```diff
@@ -58,6 +66,14 @@ def assert_schema(expected_df, checking_df, diff):
         if k not in checking_fields:
             diff.append(f"Schema field expected vs real value: {v} is not in checking value")
         elif v.dataType != checking_fields.get(k).dataType:
+            checking_type = checking_fields.get(k).dataType
+            # Dataframe loaded from an existing file will have some minor difference in schema
+            # nullable vs not nullable for ArrayType field. This block is to resolve this minor difference
```

Comment:
do you know why there is a difference?

Reply:
I don't know exactly, but it is just small detail does not affect to the correctness of the data